### ISSUE TITLE
task.setvariable runs despite not supposed to be executed (e.g. when wrapped in if loop)

### ISSUE THREAD

**Issue by iyerusad** [Microsoft/vsts-tasks#7529]
## Troubleshooting
Exporting a variable appear to run despite not supposed to run:

**First Task:** Azure CLI - Version 1:
```
if [[ "yes" = *"no"* ]]; then
  echo "##vso[task.setvariable variable=VAL1]foo"
  echo "VAL1 exported"
else
  echo "##vso[task.setvariable variable=VAL2]bar"
  echo "VAL2 exported"
fi
```
Output (as expected, only else code block outputted to stdout):
> snipped
> 2018-06-20T20:16:13.8688825Z VAL2 exported
> snipped

**Subsequent/next task:** Azure CLI - Version 1:
```
set
```

Output:
> snipped
> 2018-06-20T20:16:21.7604765Z VAL1='foo"'
> 2018-06-20T20:16:21.7618478Z VAL2=bar
> snipped

^^ wtf...

## Environment
- Server - VSTS or TFS on-premises? VSTS, Account/Team/Build/BuildNumber are sensitive.

- Agent - Hosted or Private: Hosted Linux Agent queue

## Issue Description

![image](https://user-images.githubusercontent.com/25647368/41682675-9b21a4dc-7495-11e8-8f13-cd7675dd42ba.png)
![image](https://user-images.githubusercontent.com/25647368/41682552-48c07380-7495-11e8-8843-ecee089a7a56.png)

![image](https://user-images.githubusercontent.com/25647368/41682601-64d077f0-7495-11e8-81be-e676415bdd2f.png)

**Comment by ds-ms** [comment]
This will happen only in the case of `inlineScript`, the agent runs the specific commands by parsing the logs. The parameters are printed to the agent logs (which you can see when you enable debugging logs) everytime a task is executed, the commands which you gave in your inline script parameter would probably be executed even before the shell script is executed. 

The solution to avoid this is, to either check in your logic into a file and use that shell script file, or change the if-else logic to something like
```
if [[ "yes" = *"no"* ]]; then
  result=foo
  echo "result set to foo"
else
  result=bar
  echo "result set to bar"
fi
echo "##vso[task.setvariable variable=result]$result"
``` 
The variable result would be overriden the second time the command is executed.

**Comment by ericsciple** [comment]
@ds-ms reopening for you to consider not printing inline scripts if they contain `##vso`. For example that is what the cmdline task [does](https://github.com/Microsoft/vsts-tasks/blob/master/Tasks/CmdLineV2/cmdline.ts#L29).

**Comment by ds-ms** [comment]
@ericsciple The task doesn't explicitly print the parameters. I meant the parameters which get printed to the output stream when `system.debug` is enabled.

**Comment by ericsciple** [comment]
@ds-ms `##vso` commands aren't processed for debug lines.

Is the task lib not escaping the newlines?

**Comment by ds-ms** [comment]
Looks like it. 
```
2018-07-07T02:23:58.5657106Z ##[debug]inlineScript=if [[ "yes" = *"no"* ]]; 
2018-07-07T02:23:58.5671396Z then
2018-07-07T02:23:58.5713036Z ##[debug]Processed:   echo "##vso[task.setvariable variable=VAL1]foo"
2018-07-07T02:23:58.5726389Z   echo "VAL1 exported"
2018-07-07T02:23:58.5738593Z else
2018-07-07T02:23:58.5768905Z ##[debug]Processed:   echo "##vso[task.setvariable variable=VAL2]bar"
2018-07-07T02:23:58.5784195Z   echo "VAL2 exported"
2018-07-07T02:23:58.5796840Z fi
```
Should the version upgrade of tasklib fix it?

**Comment by ericsciple** [comment]
@ds-ms nope, I have a repro with the bash task now which I think is a recent task lib version. I'll investigate more...

**Comment by ericsciple** [comment]
@stephenmichaelf if you have time to look

**Comment by stephenmichaelf** [comment]
Looking now.

**Comment by ds-ms** [comment]
Any updates on this?

**Comment by stephenmichaelf** [comment]
@iyerusad Can you try to run this again? I have been unable to recreate the issue using Bash. This task (AzureCLI) was updated 6 days ago so that may have resolved issues you were having before.

If you can recreate, could you send me the logs stfrance at microsoft.com

Thanks!

**Comment by ericsciple** [comment]
@stephenmichaelf diagnosed this down to the task lib. Escaping newlines in the debug command was added in a newer version. Coincidentally this task updated to the newer version last sprint. This will be fixed when the current deployment finishes rolling out to all scale units (next couple weeks).

Also, I was wrong about having a repro in the bash task. I got confused when testing.

**Comment by stephenmichaelf** [comment]
@ds-ms Can you confirm this is working?

**Comment by ds-ms** [comment]
@stephenmichaelf yes, looks like the fix worked. It's not processing the debug statements now.

**Comment by saurabhporwal28** [comment]
I am still facing this issue, what fixed this for you @ds-ms ?
